src/useradd.c

```diff
@@ -805,12 +814,54 @@ static int get_groups (char *list)
 	 * Any errors in finding group names are fatal
 	 */
 	if (0 != errors) {
+		close_group_files ();
```

Comment:
And here as well I'd avoid the duplication and move the close/unlock before the if().

Reply:
As you say it makes more sense to close the files before the if(), in order to minimize future errors. I have just pushed the changes.